frontend/packages/pipelines-plugin/src/components/repository/__tests__/RepositoryLinkList.spec.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { PipeLineRunWithRepoMetadata } from '../../../test-data/pipeline-data';
+import RepositoryLinkList from '../RepositoryLinkList';
+
+describe('RepositoryLinkList', () => {
+  it('should not render dl when no repository related labels exists', () => {
+    const repositoryWrapper = shallow(
+      <RepositoryLinkList pipelineRun={PipeLineRunWithRepoMetadata[0]} />,
+    );
+    expect(repositoryWrapper.find('dl').exists()).toBe(false);
+  });
+
+  it('should return correct details when repository related labels exists', () => {
+    const repositoryWrapper = shallow(
+      <RepositoryLinkList pipelineRun={PipeLineRunWithRepoMetadata[1]} />,
+    );
+    expect(repositoryWrapper.find('dl').exists()).toBe(true);
+    const repoData = repositoryWrapper.find('dl').find('dd');
+
+    expect(repoData.length).toBe(4);
+  });
+
+  it('should only the required data', () => {
+    const repositoryWrapper = shallow(
+      <RepositoryLinkList pipelineRun={PipeLineRunWithRepoMetadata[2]} />,
+    );
+    expect(repositoryWrapper.find('dl').exists()).toBe(true);
+    const repoData = repositoryWrapper.find('dl').find('dd');
+    expect(repoData.length).toBe(2);
+  });
+});
```

Comment:
In this test, If the `dl` or `dd` is changed to some other component in future, then we will have a broken test.

Few things to keep in my mind while writing unit test cases 

1. Avoid using the components directly to find and assert. 
2. If there is need to test the fields existence, then it must have a data-test attribute and use that attribute to find the field rather than directly referencing the element like `'find(dl)'`
3. Add test cases for all the conditionally renderings (in your case its all the repo labels).

eg: If Branch is set in the label, then expect the Branch Field is shown in the UI
```
 {plrLabels[RepositoryLabels[RepositoryFields.BRANCH]] && (
        <>
          <dt  data-test="branch-title">{t('pipelines-plugin~Branch')}</dt>
          <dd>{plrLabels[RepositoryLabels[RepositoryFields.BRANCH]]}</dd>
        <>
      )}
  ```

Reply:
I've updated with individual data-test-id's. Thanks for the review once again :)

Reply:
It should be data-test instead of data-test-id, that is what we follow for all the e2e tests also
Ref: https://github.com/openshift/console/blob/master/frontend/packages/integration-tests-cypress/support/selectors.ts#L47-L50

Reply:
pushed